### ISSUE TITLE
provider/openstack: Instances booted from a volume with an additional volume attached are not idempotent

### ISSUE THREAD

**Issue by robcoward** [hashicorp/terraform#7297]
### Terraform Version
```
> terraform -v
Terraform v0.6.16
```

### Affected Resource(s)
- openstack_compute_instance_v2
- openstack_blockstorage_volume_v1

### Terraform Configuration Files
```hcl
resource "openstack_blockstorage_volume_v1" "db_system" {
  name = "db_system"
  size = 30
  image_id = "${var.IMAGE_ID}"
  description = "System volume for MySQL"
}

resource "openstack_blockstorage_volume_v1" "db_data" {
  region = ""
  name = "db_data"
  description = "data volume for MySQL"
  size = 50
}

resource "openstack_compute_instance_v2" "database" {
  name            = "db01"
  flavor_name     = "${var.INSTANCE_TYPE}"
  key_pair        = "${openstack_compute_keypair_v2.ssh-keypair.name}"
  security_groups = ["${openstack_networking_secgroup_v2.local_ssh.name}",
                     "${openstack_networking_secgroup_v2.local_mysql.name}"]

  block_device {
    uuid = "${openstack_blockstorage_volume_v1.db_system.id}"
    source_type = "volume"
    boot_index = 0
    volume_size = "${openstack_blockstorage_volume_v1.db_system.size}"
    destination_type = "volume"
    delete_on_termination = true
  }

  volume {
    volume_id = "${openstack_blockstorage_volume_v1.db_data.id}"
  }

  network {
    name = "${openstack_networking_network_v2.dmz.name}"
  }
}
```

### Debug Output
https://gist.github.com/robcoward/a1c448c554a57b22da4fd2a38321d933

### Expected Behavior
When the above config is applied the first time, 2 new volumes are created, one from an image, and the instance is booted from the volume.
When running a subsequent 'plan', it should report no changes are needed. 
When running a subsequent 'apply', no changes should be applied.

### Actual Behavior
On the first 'apply', the instance and it's 2 volumes are created successfully. When running a plan or subsequent apply, terraform believes that something requires an update and tries to change the block devices on the instance, with the operation failing with a message from openstack saying that the root volume cannot be detached.

From the 'plan':
```
~ openstack_compute_instance_v2.database
    volume.#:                    "2" => "1"
    volume.4060898234.volume_id: "636d8e2c-ba43-455b-be54-25575d5f1437" => "636d8e2c-ba43-455b-be54-25575d5f1437"


Plan: 0 to add, 1 to change, 0 to destroy.
```
From the 'apply':
```
  volume.4060898234.volume_id: "636d8e2c-ba43-455b-be54-25575d5f1437" => "636d8e2c-ba43-455b-be54-25575d5f1437"
Error applying plan:

1 error(s) occurred:

* openstack_compute_instance_v2.database: Expected HTTP response code [202 204] when accessing [DELETE https://beta.openstack.skyscapecloud.com:13774/v2.1/51c7ec94d0a1420f9f55113dadbf2ec4/servers/831d9aa9-c310-4020-8857-994fc7a6fe1b/os-volume_attachments/1b722633-aad1-4bed-b4f5-cf13d857b7ea], but got 403 instead
{"forbidden": {"message": "Can't detach root device volume", "code": 403}}
```

### Steps to Reproduce
Using the sample config above:
1. `terraform apply`
2. `terraform plan`
3. `terraform apply`

### Important Factoids
When I remove the 2nd volume from the instance, so it only have the bootable volume, everything works as expected - it seems that the number of volumes actually attached to the instance being reported is incorrect when multiple volumes are attached.

### References
Raised initially on google forum as post https://groups.google.com/forum/#!topic/terraform-tool/dkmRUVLFFeQ. 
@jtopjian asked for it to be raised as an issue.

**Comment by jtopjian** [comment]
@robcoward Thank you for the report!

**Comment by GiovanniPaoloGibilisco** [comment]
I incurred in the same issue,

**Comment by GiovanniPaoloGibilisco** [comment]
The problem was there also on version 0.6.14. 

It also occurs then the root volume is created within the definition of the VM as in: 

```
resource "openstack_blockstorage_volume_v1" "additional_volume_1" {
  name = "additional_volume_1"
  size = 5
}

resource "openstack_compute_instance_v2" "test_name" {
  name = "test_name"
  flavor_id = "3"
  key_pair = "test-keypair"
  security_groups = ["default"]

  block_device {
    uuid = "90ee49d8-1c18-45ca-8a72-14ffd120521e"
    source_type = "image"
    volume_size = 10
    boot_index = 0
    destination_type = "volume"
    delete_on_termination = false
  }

  volume {
    volume_id = "${openstack_blockstorage_volume_v1.additional_volume_1.id}"
  }

  metadata {
    application = "app1"
  }

  network {
    name = "network"
  }
}
```

**Comment by marcomusso** [comment]
I've got a similar problem with two attached volumes (not boot) that could be related.
After an `apply` and with no changes in `.tf` files `terraform plan` wants to swap block devices:

```
~ openstack_compute_instance_v2.master
    volume.3445100101.device: "/dev/sdb" => "/dev/sdc"
    volume.549565416.device:  "/dev/sdc" => "/dev/sdb"
```

**Comment by jtopjian** [comment]
Hi all,

I have a proposed fix for this at #7649. It's more of a workaround than a proper fix: if an attached volume is attached as device `/dev/vda`, `/dev/sda`, or `/dev/xda`, it won't be reported in the list of volumes.

If anyone knows of any more device names, please let me know.

**Comment by minhchuduc** [comment]
@jtopjian you could add `/dev/xvda`
http://askubuntu.com/questions/166083/what-is-the-dev-xvda1-device

**Comment by jtopjian** [comment]
@minhchuduc Added! Thanks 😄

**Comment by robcoward** [comment]
I can confirm that this is now fixed in the 0.7 release. Thanks @jtopjian
